samples/GenericWebHost/WebHostExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.ObjectPool;
+
+namespace GenericWebHost
+{
+    public static class WebHostExtensions
+    {
+        public static IHostBuilder ConfigureWebHost(this IHostBuilder builder, Action<HostBuilderContext, IApplicationBuilder> configureApp)
```

Comment:
Do you think we should ship this in the Hosting package?

Reply:
Not yet. This was just a sample for how we might redesign WebHost in the future.

Reply:
I think that's a second set of changes anyway.